### ISSUE TITLE
Can't load any SLANG shader preset from QT interface

### ISSUE THREAD

**Issue by ofry** [libretro/RetroArch#10984]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

Can't load any SLANG shader preset from QT interface

### Expected behavior

Shader preset should load into window.

### Actual behavior

Nothing happened, shader window stays without any shaders.

### Steps to reproduce the bug

1. Load Retroarch
2. Switch video driver to "glcore"
3. Restart Retroarch.
4. Open Desktop QT UI.
5. Go to shaders.
6. Open any preset, for example, "mdapt.slangp"
7. Nothing happened, In console window shows message:
```
[INFO] [Qt]: Reloading shader parameters.
```

### Bisect Results

[Try to bisect and tell us when this started happening]

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 6df4e7778f

### Environment information

- OS: Windows 8.1 Enterprise x64
- Compiler:

**Comment by hizzlekizzle** [comment]
@HyperspaceMadness Any thoughts?

I would expect it to be something in here: https://github.com/libretro/RetroArch/blob/master/ui/drivers/qt/shaderparamsdialog.cpp

But nothing is jumping out at me.

**Comment by bslenul** [comment]
> [INFO] [Qt]: Reloading shader parameters.

Nothing shader-related before this in the logs? Are you sure you have a game loaded before trying to load the preset?

**Comment by Tatsuya79** [comment]
![image](https://user-images.githubusercontent.com/6266038/86900349-877ae280-c10b-11ea-9445-ca6e75919cf4.png)

Working fine here win7 x64.

**Comment by ofry** [comment]
@Tatsuya79 I did it BEFORE loading any game.

**Comment by bslenul** [comment]
OK, that's what I thought then, you have to load a game first ;) It's a bit confusing because with Desktop Menu the shaders options are always available, even without content loaded. But yeah, you need to load a game first!

**Comment by ofry** [comment]
@bslenul But why I can't edit retroarch.slangp (global shader preset)? It was loaded even before game starts.

**Comment by bslenul** [comment]
Not sure, for me it's empty until I load content. By curiosity I just tried with old stables, up to 1.7.7 it was actually possible to manage shaders without content loaded (I didn't even know that :p), after that it was not possible anymore.

There were probably a ton of commits between 1.7.7 and 1.7.8 so I'm not sure I want to bisect that one :x

**Comment by bslenul** [comment]
> There were probably a ton of commits between 1.7.7 and 1.7.8 so I'm not sure I want to bisect that one :x

Well, I had time to waste so I did it anyway: 00299cf283d7d6f935d48e2623e0fd88d3c727d3 is the 1st commit where it's not possible anymore to manage shaders from Desktop Menu without content loaded.

**Comment by hizzlekizzle** [comment]
Yeah, that looks intentional. That is, I think it was loading the shaders prematurely, which can take a very long time for large shaders (especially if you're loading them over a network, like I frequently do).
